### ISSUE TITLE
Improve around Accessibility Areas in React-Beta Docsite

### ISSUE THREAD

**Issue by sohamsshah** [reactjs/reactjs.org#3969]
In the latest beta documentation for React, there are some accessibility issues which are persisting.

Some of them are (from lighthouse report):
- Missing `aria-labels` in view elements
- Links do not have descriptive text

I am willing to work and contribute to this and will try to improve upon these areas to the best so that the React Documentation site is fully accessible to everyone on the internet.

P.S. I am a first-time-contributor to React and very excited to make this contribution. Thank you.
